kotlinx-coroutines-core/common/test/TestBase.common.kt

```diff
@@ -20,6 +22,10 @@ public expect val stressTestMultiplier: Int
 @Suppress("NO_ACTUAL_FOR_EXPECT")
 public expect class TestResult
 
+public expect val isNative: Boolean
+// "Speedup" native stress tests
+public val stressTestNativeDivisor = if (isNative) 10 else 1
```

Comment:
I get that some tests should have fewer iterations on Native. However, the existence of this constant seems to imply that *all* tests will run with fewer iterations. And this is true: when `isStressTest` is true, `stressTestMultiplier` is *also* lower than on the JVM. So, we have three disparate mechanisms for dealing with the number of stress test iterations:
* `stressTestMultiplier` when `isStressTest` is true;
* For some tests, `stressTestNativeDivisor`;
* For some other tests, manual checks and overrides.

Also, I see that `stressTestNativeDivisor` is only used in a few places. Can we maybe simplify this system a bit by getting rid of this constant and instead, doing manual checks in the few tests that it is needed in?

Reply:
8 places (potentially there will be more) looks like quite a lot.
Maybe we'll rename it to something more explicit? E.g. `nativeSpeciticSlowTestDivisor` with a comment?

Reply:
Explicitness would help, but I just don't understand what we achieve by introducing this constant (which, I assume, doesn't have inherent meaning and is just a heuristic) in the first place.
```kotlin
val n = 10_000 * stressTestMultiplier / stressTestNativeDivisor
val n = (if (isNative) 1000 else 10_000) * stressTestMultiplier
```
To me, the second version is easier to grasp and is also easier to tweak on a per-test basis.